proxy/http/HttpSM.cc

```diff
@@ -5135,8 +5135,13 @@ HttpSM::get_outbound_sni() const
   swoc::TextView policy{t_state.txn_conf->ssl_client_sni_policy, swoc::TextView::npos};
 
   if (_ua.get_txn()) {
-    if (const NetVConnection *netvc = _ua.get_txn()->get_netvc(); netvc->options.outbound_sni_policy) {
-      policy.assign(netvc->options.outbound_sni_policy.get(), swoc::TextView::npos);
+    if (auto *netvc = _ua.get_txn()->get_netvc(); netvc) {
+      if (auto *snis = netvc->get_service<TLSSNISupport>(); snis && snis->hints_from_sni.outbound_sni_policy.has_value()) {
+        netvc->options.outbound_sni_policy = snis->hints_from_sni.outbound_sni_policy.value();
+      }
+      if (netvc->options.outbound_sni_policy) {
+        policy.assign(netvc->options.outbound_sni_policy.get(), swoc::TextView::npos);
+      }
```

Comment:
Should we just get rid of outbound_sni_policy from the netvc->options?  And just pull from the hints_from_sni directly?  It does not look like netvc->options.outbound_sni_policy is used otherwise.

Reply:
Good catch. I thought it's overwritten by SNI config.